ONSdigital/zebedee
# Fix bug where Welsh content was overwritten by English changes

### What

If English changes are present in a collection and we try to get the equivalent Welsh content, it should not retrieve the English changes but the published Welsh content.

Override the `resolveDataFilePath` method when working with collections so that it won't find the English version in there but go and find the published content [further down the line](https://github.com/ONSdigital/zebedee/blob/4a1c3797ad13c6b17ab2768914eaa8b3d8e025f3/zebedee-reader/src/main/java/com/github/onsdigital/zebedee/reader/api/ReadRequestHandler.java#L138).

Additionally, `ReadRequestHandler` will now try to get the English version if no content is found. This is so that we can find the English content in the collection when creating a new page in Welsh (that is not already published).

Fixes https://trello.com/c/GpcNQzC3/1163-english-content-overwriting-welsh-content-in-florence

### How to review

- In Florence create content (for example a bulletin) in both English and Welsh
- Approve and publish the collection
- Create a new collection
- Find and edit the previous content in English. Save
- Toggle to Welsh language and find and edit the same content.

Expected: It should load the original Welsh content and not the amended English content in the collection.

### Who can review

Anyone
